packages/backend-defaults/src/PackageDiscoveryService.ts

```diff
@@ -137,6 +137,10 @@ export class PackageDiscoveryService {
         if (isError(error) && error.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
           continue; // Skip packages that don't export package.json - they can't be Backstage packages
         }
+        // Handle packages that cannot be found (e.g., optional dependencies)
+        if (isError(error) && error.code === 'MODULE_NOT_FOUND') {
```

Comment:
No checking whether it IS optional or not?

Reply:
+1, I feel we would need a more strict check for this. There's a bunch of other ways this could happen that you'd want to catch